### ISSUE TITLE
node-sass silently fails when importing files

### ISSUE THREAD

**Issue by phoenixenero** [sass/node-sass#1315]
First got this issue when my Gulp Sass tasks suddenly stopped finishing. I then tried to compile my Sass with `node-sass` cli directly but it outputs nothing at all.

I later isolated the issue to an import for Susy. The file which had the `@import` line, is located in:

```
config/structure/_susy.scss
```

Relative to the main `style.scss` which imports everything in my project.

When I changed the `@import` to `@import 'sosy'`, it outputs an error saying it doesn't exist (as expected.)

Yet when I `@import 'susy'` inside my `_susy.scss` partial, it doesn't output anything at all. No errors, no rendered file.

Then I think: maybe the name of the file &mdash; `_susy.scss`&mdash;  is the problem?

Sure enough, when I renamed my Susy config file from `_susy.scss` to `_susyconf.scss` and re-ran it, everything works!

* * *

However it took me quite a while to find a problem. This should've outputted as an error, or basically not silently failing while I'm working.

**TL;DR**: Doing an `@import bar` inside a file `_bar.scss` will silently fail with no errors.

**Edit:** Made it clearer.

**Comment by saper** [comment]
Can you try to reproduce the problem using plain node-sass without using gulp? Also "it errors" is not enough information to diagnose the problem. The information in https://github.com/sass/node-sass/blob/master/TROUBLESHOOTING.md might help to diagnose the problem.

**Comment by phoenixenero** [comment]
Sorry if I seemed unclear.

I *did* use `node-sass` commandline after troubleshooting gulp. And the problem I had is not that it errors, but that it **doesn't error**.

**Comment by phoenixenero** [comment]
I had already solved my problem, but for those who would inevitably name their Susy config partials `_susy.scss` and import Susy in that file, it would be very irritating to see no feedback.

**Comment by phoenixenero** [comment]
Here's the `node-sass -v` output if it's necessary:

```
node-sass       3.4.2   (Wrapper)       [JavaScript]
libsass         3.3.2   (Sass Compiler) [C/C++]     
```

**Comment by saper** [comment]
I am sorry, 
I still cannot get how to work on this problem. Can you publish a short, self-contained example how to reproduce the problem with node-sass?

**Comment by saper** [comment]
I have tried this:

=== start.scss===
````
@import 'susy';
````
=== _susy.scss ===
````
@import 'susy';
````

it seems to core dump with after a while with "invalid instruction", the backtrace suggests that libsass loops endlessly including the file.

**Comment by saper** [comment]
(confirmed with pure `sassc`).

**Comment by saper** [comment]
I have filed https://github.com/sass/libsass/issues/1801 for this.

**Comment by xzyfer** [comment]
Thanks for digging into this @saper. We'll track this in LibSass.
